mongo/collection.go

```diff
@@ -1543,6 +1837,49 @@ func (coll *Collection) FindOneAndDelete(ctx context.Context, filter interface{}
 	return coll.findAndModify(ctx, op)
 }
 
+// mergeFindOneAndReplaceOptions combines the given FindOneAndReplaceOptions instances into a single
+// FindOneAndReplaceOptions in a last-one-wins fashion.
```

Comment:
Optional: see similar for explanation

```suggestion
// FindOneAndReplaceOptions in a last-property-wins fashion.
```